File: lab2/lab2.py


x=int(input("Please enter a positive integer greater than 9: "))

print(x, end="")
count=0
while x > 9:
    currentSum=0
    tempX=x
    while tempX>0:
        currentSum+=(tempX%10)
        tempX//=10

    x=currentSum
    count+=1
    print(" → ",currentSum,end="")

print("\nFinal value: ", x)
print("Total steps: ", count)

#question 2
from operator import contains

while True:
    x = int(input("Please enter a number between 10 and 100: "))

    if 10 <= x <= 100:
        break
    print("Invalid input. ", end="")

fizz=0
buzz=0
fizzbuzz=0
for a in range(1,x+1):

    if a % 7 == 0:
        print(f"({a} is skipped)")
        continue
    if a % 3 == 0 and a % 5 == 0:
        print("FizzBuzz")
        fizzbuzz+=1
    elif a % 3 == 0:
        print("Fizz")
        fizz+=1
    elif a % 5 == 0:
        print("Buzz")
        buzz+=1
    else:
       print(a)
print("--- Summary ---")
print("Fizz count : ",fizz)
print("Buzz count : ",buzz)
print("FizzBuzz count: ",fizzbuzz)

#question 3


x= int(input("Please enter a number between 3 and 9: "))

for i in range(1,2*x):
    k = x - abs(x - i)
    for j in range(1,k+1):
        print(j,end=" ")

    print()

File: lab2/lab2.cpp
#include <iostream>
using namespace std;

int main()
{
    int x;
    cout<<"Please enter a positive integer greater than 9: ";
    cin>>x;
    int count = 0;
    cout<<x;
    while (x>9)
    {
        int currentSum=0;
        int tempX=x;
        while (tempX>0)
        {
            currentSum+=(tempX%10);
            tempX=tempX/10;
        }
        x=currentSum;
        count++;
        cout<<" → "<<currentSum;

    }
    cout<<"\nFinal value: "<<x<<endl;
    cout<<"Total steps: "<<count<<endl;

    int y;
    cout << "Please enter a number between 10 and 100: ";
    while (true)
    {
        cin>>y;
        if (y>=10 && y<=100) break;
        cout<<"Invalid input. Please enter a number between 10 and 100: ";
    }

    int fizz=0;
    int buzz=0;
    int fizzbuzz=0;
    for (int i =1;i<=y;i++)
    {
        if (i % 7 ==0)
        {
            cout<<"("<<i<<" is skipped)"<<endl;
            continue;
        }
        if (i % 3 == 0 && i % 5 == 0)
        {
            fizzbuzz++;
            cout<<"FizzBuzz"<<endl;
        }
        else if (i % 3 == 0)
        {
            fizz++;
            cout<<"Fizz"<<endl;
        }else if (i % 5 == 0)
        {
            buzz++;
            cout<<"Buzz"<<endl;
        }else
        {
            cout<<i<<endl;
        }
    }
    cout<<"--- Summary ---"<<endl<<
        "Fizz count : "<<fizz<<endl<<
        "Buzz count : "<<buzz<<endl<<
        "FizzBuzz count: "<<fizzbuzz<<endl;


    int a;
    cout<<"Please enter a number between 3 and 9: ";
    cin>>a;
    for (int i=1; i<=2*a-1; i++)
    {
        int k = a - abs(a-i);
        for (int j=1; j<=k; j++)
        {
            cout<<j;
        }
        cout<<endl;

    }
    return 0;
}

